### PR TITLE
[Enhancement] Support trace propagation using Traceparent header and add trace for TableSink

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

This PR adds a trace propagation mechanism using the standard `traceparent` header, also adds trace propagation in publish_version, OlapTableSink and LoadChannel.

### CODE REVIEW

**Comment by decster** [comment]
Screenshot from 2022-06-17 20-21-36

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 23 / 30 (76.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/PublishVersionTask.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/common/TraceManager.java|15|16|93.75%|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|4|4|100.00%|

**Comment by decster** [comment]
run starrocks_fe_unittest

**Review by Astralidea — DISMISSED** [review]

**Comment by imay** [comment]
@decster Could you write a document for our developers to use this tool?

**Review by Linkerist — DISMISSED** [review]

File: be/src/exec/tablet_sink.cpp [hunk 1]
```diff
@@ -585,6 +588,7 @@ Status OlapTableSink::init(const TDataSink& t_sink) {
 }
 
 Status OlapTableSink::prepare(RuntimeState* state) {
+    _span->AddEvent("prepare");
```
Comment (chaoyli) [inline be/src/exec/tablet_sink.cpp:591]: Could it add query_id/load_id to trace a query or load?

File: gensrc/thrift/DataSinks.thrift [hunk 2]
```diff
@@ -146,6 +146,7 @@ struct TOlapTableSink {
     13: required Descriptors.TNodesInfo nodes_info
     14: optional i64 load_channel_timeout_s // the timeout of load channels in second
     15: optional bool is_lake_table
+    16: optional string txn_trace_parent
```
Comment (chaoyli) [inline gensrc/thrift/DataSinks.thrift:149]: Does the length of the string will too long?

Comment (decster) [reply gensrc/thrift/DataSinks.thrift:149]: no, only fixed 55 bytes

File: be/src/exec/tablet_sink.cpp (hunk 1 above)
Comment (decster) [reply be/src/exec/tablet_sink.cpp:591]: I already add `label`

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 39 / 48 (81.25%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/PublishVersionTask.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|16|18|88.89%|
|:large_blue_circle:|com/starrocks/common/TraceManager.java|15|16|93.75%|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|4|4|100.00%|

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**